cpp/src/parquet/column_writer.cc

```diff
@@ -335,6 +340,10 @@ class SerializedPageWriter : public PageWriter {
     if (meta_encryptor_ != nullptr) {
       UpdateEncryption(encryption::kColumnMetaData);
     }
+
+    // Serialized page writer does not need to adjust page offsets.
+    FinishPageIndex(/*final_position=*/0);
```

Comment:
Maybe, FinishPageIndexes (it took me looking up the actual method implementation when reading through this that this finished both, column and offsets).